drivers/docker/reconcile_cpuset.go

```diff
@@ -0,0 +1,125 @@
+//go:build linux
+
+package docker
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad/client/lib/cgutil"
+	"github.com/hashicorp/nomad/helper"
+)
+
+const (
+	cpusetReconcileInterval = 1 * time.Second
+)
+
+type CpusetFixer interface {
+	Start()
+}
+
+// cpusetFixer adjusts the cpuset.cpus cgroup value to the assigned value by Nomad.
+//
+// Due to Docker not allowing the configuration of the full cgroup path, we must
+// manually fix the cpuset values for all docker containers continuously, as the
+// values will change as tasks of any driver using reserved cores are started and
+// stopped, changing the size of the remaining shared cpu pool.
```

Comment:
Just for my clarity, in the case of https://github.com/hashicorp/nomad/issues/11705 with cgroups v2: any child processes will be in the same cgroup as their parent that Docker starts. So although they won't be pinned to the right CPU for a very brief window at startup, they'll all get moved together to the correct `cpuset` because we're not moving processes between cgroups, just changing the cgroup in place.

Is my understanding right?

Reply:
Yup! And actually this is probably worth amending the RFC and reconsidering. 

Although this behavior is most similar to the way V1 works, I don't see why it wouldn't be worth just setting the cpuset on the docker task config on initial startup now. The drawback is I don't think we have the plumbing to get the `shared` cores - the initial value would only be the `reserved` cores requested by the task resources.

Reply:
Our trading infrastructure reads the cpus in the cgroup on startup and index into the cpuset to pin threads to specific cores. If the cgroup gets modified after we've done this thread <-> core pinning, I'm not entirely sure what happens but it's bad as either threads will still be pinned to CPUs outside of the allocated cgroup or it just breaks altogether.

cf: https://github.com/hashicorp/nomad/issues/12374